### PR TITLE
Fix PYPY_VERSION_HEX/PYPY_VERSION_NUM typo

### PR DESCRIPTION
**Pull request by mattip**
Fix a typo: `s/PYPY_VERSION_HEX/PYPY_VERSION_NUM`

Along the way, drop support for PyPy < 7.0.0, which was released in April 2019. I doubt anyone can use c-extensions compiled with those versions, support was very weak.

### CODE REVIEW

File: Cython/Utility/ModuleSetupCode.c [hunk 1]
```diff
@@ -169,7 +172,7 @@
   #undef CYTHON_USE_EXC_INFO_STACK
   #define CYTHON_USE_EXC_INFO_STACK 0
   #ifndef CYTHON_UPDATE_DESCRIPTOR_DOC
-    #define CYTHON_UPDATE_DESCRIPTOR_DOC (PYPY_VERSION_HEX >= 0x07030900)
+    #define CYTHON_UPDATE_DESCRIPTOR_DOC (PYPY_VERSION_NUM >= 0x07030900)
```
Comment (da-woods) [inline Cython/Utility/ModuleSetupCode.c:175]: This is probably worth picking to 0.29.x. I'm inclined not to pick the rest on the basis that we're largely trying not to change it.

Comment (mattip) [reply Cython/Utility/ModuleSetupCode.c:175]: OK. I can drop the last commit and just leave the typo in this PR.

Comment (da-woods) [reply Cython/Utility/ModuleSetupCode.c:175]: Wasn't asking you to do anything with this PR. Was just noting that it's worth making a 0.29.x PR with just that change

Comment (mattip) [reply Cython/Utility/ModuleSetupCode.c:172]: No problem, let's consider this the 0.29.x PR :)

**Comment by da-woods** [comment]
Cherry-picked in 3a063cb43

**Comment by da-woods** [comment]
I wonder if we shouldn't have merged this - there's a few segmentation faults in the test run https://github.com/cython/cython/actions/runs/3214740819/jobs/5255340663, which suggests `CYTHON_UPDATE_DESCRIPTOR_DOC` isn't yet working on PyPy.

I'll revert it for now. Obviously we do want to fix the typo though. But possibly just to keep this feature off

**Comment by mattip** [comment]
Weird. I see CI is hanging. How can I determine which tests are segfaulting?

**Comment by da-woods** [comment]
> I see CI is hanging.

That's fairly typical - if one of the threads segfaults the CI will wait forever for it to finish.

> How can I determine which tests are segfaulting?

Not too easily unfortunately. We're using the faulthandler module which gives you a bit of feedback, but output from multiple threads is mixed together.

What I'd do is set the CI going again but with only one thread

https://github.com/cython/cython/blob/1741bac96ce49d7e31b2434dd798fa38dad660dd/Tools/ci-run.sh#L175

and it'll probably be a lot clearer.

**Comment by da-woods** [comment]
@mattip I've started #5077 - hopefully single threaded. That should point you to the first failure. I suspect all the failures are the same so that'll probably be enough!

**Comment by mattip** [comment]
By running `gdb --args pypy3 runtests.py --no-capture --no-fork -vv` and letting tests run for a while, it crashes in the pypy garbage collector, each time in a different point in the tests, but somewhere around DocTests. Does that agree with what you see in #5077?

**Comment by da-woods** [comment]
All I know from #5077 is that it crashed once in "altet2". So it doesn't disagree. But from what you say it sounds like a very general error rather than one linked to a particular test.

**Comment by da-woods** [comment]
Yeah I think the unconditional define is probably the right thing to do for the moment

**Pull request merged**